arrows/core/track_set_impl.cxx

```diff
@@ -540,27 +540,28 @@ ::set_frame_data( track_set_frame_data_sptr data,
 
 track_set_implementation_uptr
 frame_index_track_set_impl
-::clone() const
+::clone( vital::clone_type ct ) const
 {
   std::unique_ptr<frame_index_track_set_impl> the_clone =
     std::unique_ptr<frame_index_track_set_impl>(new frame_index_track_set_impl());
 
   // clone the track data
-  for (auto trk : all_tracks_)
+  for ( auto const& trk : all_tracks_ )
   {
-    the_clone->all_tracks_.insert(std::make_pair(trk.first,trk.second->clone()));
+    the_clone->all_tracks_.emplace( trk.first, trk.second->clone( ct ) );
   }
 
   // clone the frame data
-  for (auto fd : frame_data_)
+  for ( auto const& fd : frame_data_ )
   {
-    if (fd.second)
```

Comment:
Why are we dropping the check on `fd.second`, what if the data is null?

Reply:
AFAICT that can only happen if someone subclasses `frame_index_track_set_impl` and inserts a null datum into the map. Otherwise, the only place data is inserted into `frame_data_` is `frame_index_track_set_impl::set_frame_data`, which only performs the insertion if the frame data is non-null (otherwise it removes entries from the map).

...and if we're going to be paranoid, we should be checking `all_tracks_` also, but we aren't/weren't.

Reply:
alright, that's fine then.  I think those guards were added because at one point the data could be null and it was crashing, but then we must have also added the checks to prevent null data in the first place.  So I guess it's okay.